retrofit/src/test/java/retrofit2/NonFatalError.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package retrofit2;
+
+final class NonFatalError extends Error {
```

Comment:
Hmm, what does error-prone complain about here?

Reply:
the test code was basically
```java
try {
  throw new Error();
  fail();
} catch (Error e) {
}
```
Since `fail()` will throw `AssertionError`, it will be caught by the `catch` block even if the behavior under test fails. By changing to a specific subtype and catching only that subtype we allow an `AssertionError` to propagate and correctly fail the test.

Reply:
Nice!